java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -135,6 +136,11 @@ public ExplodedGraphWalker(JavaFileScannerContext context) {
     this.cleanup = cleanup;
   }
 
+  private ExplodedGraphWalker(JavaFileScannerContext context, ConditionAlwaysTrueOrFalseCheck alwaysTrueOrFalseChecker, List<SECheck> seChecks) {
```

Comment:
Given the use of factory, ALL the other constructors should be private adn/or package with @VisibleForTesting annotation.